sunpy/map/mapbase.py

```diff
@@ -243,7 +246,45 @@ def __repr__(self):
                                refpix=u.Quantity(self.reference_pixel),
                                refcoord=u.Quantity((self.reference_coordinate.data.lon,
                                                     self.reference_coordinate.data.lat)),
-                               tmf=TIME_FORMAT) + self.data.__repr__()
+                               tmf=TIME_FORMAT)
+
+    def __repr__(self):
+        return self._text_summary() + "\n" + self.data.__repr__()
```

Comment:
I have been chatting to Dan about this in NDCube recently, and this is a good time to do it here as well. I strongly feel that all our repr's should have the `object.__repr__` at the start, so it's clear what the type is and what the id is.

 ```suggestion
        return object.__repr__(self) + "\n" + self._text_summary() + "\n" + self.data.__repr__()
```

Reply:
Done